src/gordon_gcp/plugins/_utils.py

```diff
@@ -15,11 +15,74 @@
 # limitations under the License.
 """Common utils shared among plugins."""
 
-
+import functools
 import logging
 
+from gordon_gcp import exceptions
+
 
 class GEventMessageLogger(logging.LoggerAdapter):
     def process(self, log_msg, kwargs):
         log = f'[msg-{self.extra["msg_id"]}]: {log_msg}'
         return log, kwargs
+
+
+# TODO: should this be named something like `handle_errors_async`?
```

Comment:
@econchick so to change it from route_result to handle_errors_async?

Reply:
If you are going to rename it to include async (and i'm not sure this is a naming pattern, but i think it might be useful), I think handle errors is too vague.  `async_route_result` is ok, but maybe `async_route_result_to_channel` to be a little more descriptive?